04_linked_list/lib/linked_list.js

```diff
@@ -132,15 +132,34 @@ class LinkedList {
   }
 
   //de-serialize method - turns serialized linked list back into linked list  - BigO = O(1)
-  static deSerialize(serializedList) {
-    let deSerializedList = serializedList.toString();
-    deSerializedList = JSON.parse(deSerializedList);
+  static deserialize(serializedList) {
+    let deserializedList = serializedList.toString();
+    deserializedList = JSON.parse(deserializedList);
     let newLinkedList = new LinkedList();
-    newLinkedList.length = deSerializedList.length;
-    newLinkedList.head = deSerializedList.head;
-    newLinkedList.tail = deSerializedList.tail;
+    newLinkedList.length = deserializedList.length;
+    newLinkedList.head = deserializedList.head;
+    newLinkedList.tail = deserializedList.tail;
     return newLinkedList;
   }
+  //mergeLists
+  static mergeLists(listOne, listTwo){
+    let startLength = listOne.length;
+    let current = listOne.head;
+    
+    let i = 0;
+    while (i < startLength && listTwo.head !== null){
+      listOne.insertAfter(current.value, listTwo.head.value);
+      listTwo.head = listTwo.head.next;
+      current = current.next.next;
+      i++;
+    }
+    console.log(listOne.head);
+
+    if (listTwo.head){
+      listOne.tail = listTwo.head;
+    }
+    return listOne.head, listOne.length, listOne.tail;

```

Comment:
These are the lab requirements:
```
Zip the two linked lists together into one so that the nodes alternate between the two lists and return a reference to the head of the zipped list.
```
Returning the exact value(s) that are required is extremely important when writing code. In this case, mergeLists() should return only a reference to the head of the zipped list. Nothing more and nothing less. Changing a function's signature (input/output) is a big deal in the coding world as it makes the function become fundamentally different. Implementation may change but not affect the user, while changing the input/output definitely will affect the user.